near/omni-bridge/src/lib.rs

```diff
@@ -517,11 +547,147 @@ impl Contract {
+                };
+                PromiseOrValue::Promise(
+                    Self::check_or_pay_ft_storage(
+                        &deposit_action,
+                        &mut NearToken::from_yoctonear(storage_deposit_amount),
+                    )
+                    .then(
+                        Self::ext(env::current_account_id())
+                            .with_static_gas(FAST_TRANSFER_CALLBACK_GAS)
+                            .fast_fin_transfer_to_near_callback(fast_transfer, sender_id),
+                    ),
+                )
+            }
+            _ => {
+                self.fast_fin_transfer_to_other_chain(fast_transfer, sender_id);
+                PromiseOrValue::Value(U128(0))
+            }
+        }
+    }
+
+    #[private]
+    pub fn fast_fin_transfer_to_near_callback(
+        &mut self,
+        #[serializer(borsh)] fast_transfer: FastTransfer,
+        #[serializer(borsh)] relayer_id: AccountId,
+    ) -> Promise {
+        require!(
+            Self::check_storage_balance_result(0),
+            "STORAGE_ERR: The transfer recipient is omitted"
+        );
+
+        let OmniAddress::Near(recipient) = fast_transfer.recipient.clone() else {
+            env::panic_str("ERR_INVALID_STATE")
+        };
+
+        let required_balance = self
+            .add_fast_transfer(&fast_transfer, &relayer_id)
+            .saturating_add(ONE_YOCTO);
+        self.update_storage_balance(relayer_id, required_balance, NearToken::from_yoctonear(0));
+
+        env::log_str(
+            &OmniBridgeEvent::FastTransferEvent {
+                fast_transfer: fast_transfer.clone(),
+                new_transfer_id: None,
+            }
+            .to_log_string(),
+        );
+
+        self.send_tokens(
+            fast_transfer.token_id,
+            recipient,
+            U128(fast_transfer.amount.0 - fast_transfer.fee.fee.0),
+            fast_transfer.msg,
+        )
+        .then(
+            Self::ext(env::current_account_id())
+                .with_static_gas(RESOLVE_TRANSFER_GAS)
+                .resolve_transfer(fast_transfer.amount),
+        )
+    }
+
+    fn fast_fin_transfer_to_other_chain(
+        &mut self,
+        fast_transfer: FastTransfer,
+        relayer_id: AccountId,
+    ) {
+        if self.is_transfer_finalised(fast_transfer.transfer_id) {
+            env::panic_str("ERR_TRANSFER_ALREADY_FINALISED");
+        }
+
+        let mut required_balance = self.add_fast_transfer(&fast_transfer, &relayer_id);
+
+        let destination_nonce =
+            self.get_next_destination_nonce(fast_transfer.recipient.get_chain());
+        self.current_origin_nonce += 1;
+
+        let transfer_message = TransferMessage {
+            origin_nonce: self.current_origin_nonce,
+            token: OmniAddress::Near(fast_transfer.token_id.clone()),
+            amount: fast_transfer.amount.clone(),
+            recipient: fast_transfer.recipient.clone(),
+            fee: fast_transfer.fee.clone(),
```

Comment:
What about creating an invalid TransferMessage with a very very large fee? Finalizing it on the destination chain, and then claiming our excessively large fee. In that case, a lot of Ether, for example, would be minted for the relayer without reason.